server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/GetSnapshotsRequest.java

```diff
@@ -319,10 +321,24 @@ public static SortBy of(String value) {
 
         private final String value;
 
+        private final String repoName;
+
         private final String snapshotName;
 
         After(StreamInput in) throws IOException {
-            this(in.readString(), in.readString());
+            this(in.readString(), in.readString(), in.readString());
+        }
+
+        public static After fromParam(String param) {
+            final String[] parts = new String(Base64.getDecoder().decode(param), StandardCharsets.UTF_8).split(",");
+            if (parts.length != 3) {
+                throw new IllegalArgumentException(
+                    "after param must be base64 encoded and of the form ${sort_value},${repository_name},${snapshot_name} but was ["
```

Comment:
No need to spell out the format here IMO

```suggestion
                    "invalid ?after parameter ["
```

Reply:
++